torch/_dynamo/skipfiles.py

```diff
@@ -147,6 +147,12 @@ def _module_dir(m: types.ModuleType):
         torch._refs.nn.functional.__file__,
     }
 
+# Skip quantization patterns
+# TODO: How to make this less brittle without introducing circular dependencies?
+FILENAME_ALLOWLIST |= {
+    _module_dir(torch) + "ao/quantization/_pt2e/qat_utils.py",
+}
+
```

Comment:
I would suggest you put this under:
https://github.com/pytorch/pytorch/blob/ebd4c165ffa586491dac51c555b8bffd1f844b5a/torch/_dynamo/skipfiles.py#L242-L245
This is where we define the rules if tracing through torch internal code.

Reply:
Looks like `allow_torch` is False when I call `torchdynamo.export` however, so we'll skip this case:
https://github.com/pytorch/pytorch/blob/01a7f24f7367f6d13ffb63f69cc6523583187620/torch/_dynamo/skipfiles.py#L199-L200
How can I make it work?

Reply:
Sounds good. However, looks like `allow_torch=False` when I call `torchdynamo.export`, so it'll skip this case:
https://github.com/pytorch/pytorch/blob/01a7f24f7367f6d13ffb63f69cc6523583187620/torch/_dynamo/skipfiles.py#L199-L200
How can I make it work?

Reply:
Oh, you can append your module here to allow inline/tracing:
https://github.com/pytorch/pytorch/blob/95621b3c2e8bc0f1d7f88fb0bcb3e3e56a0ff961/torch/_dynamo/config.py#L126-L135

Reply:
Hmm, this didn't seem to work. I tried adding `torch.ao.quantization` here and I can't export modules defined in this namespace. The other problem is I can't specify a tighter namespace (e.g. `torch.ao.quantization.pt2e.qat_utils`) due to circular imports.

Reply:
Hmm, ```torch.ao.nn``` is here, so generally if you want the PyTorch ao builtin modules can be inlined/traced by dynamo, it should be there. Do you think it makes sense to move your module under ```torch.ao.nn```? This would make the structure more clear and easy to maintain.

Reply:
I tried moving it to `torch.ao.nn` and I'm getting the same error when I export:
```
AssertionError: Failed to produce a graph during tracing. Tracing through 'f' must produce a single graph.
```
I feel `skipfiles_inline_module_allowlist` isn't working as we expect here.

Reply:
Oh, that indicates a circular imports during export. I think that's a bug if ```torch.ao.nn``` doesn't work, I'll check it and circle back to you.

Reply:
@andrewor14 Sorry for late response, was really busy last week, but finally figured it out.

I can reproduce your issue with https://gist.github.com/yanboliang/dcfa8c0e7f4249a2961630b8e98045d0.
Look at this repro: it passed with dynamo, but failed with export. The reason is export doesn't allowed None graph, but dynamo is allowed.
Why it produced None graph? because we skipped the trace of these modules. We put a ```call_module``` op in the FX graph at dynamo level and hand it over to AOT autograd to unroll/flatten these __builtin__ modules, as we can ensure they are safely traced by AOT autograd and make dynamo level simple to handle this case.
However, you are trying to allowed dynamo to trace through the modules under ```torch.ao.nn.*```, can you explain a bit more on why you have to do this?
Take a step back, if there is a strong reason you must to do this, I think the way is follow what we did for ```_prims``` and ```_refs``` at:
https://github.com/pytorch/pytorch/blob/be0b12ece576c86c5f059d15a64dcba0eb886ddd/torch/_dynamo/skipfiles.py#L142-L150

For the above repro I have, simply adding the following lines can make dynamo trace through the ```ao``` Linear module.
```
FILENAME_ALLOWLIST |= {torch.ao.nn.qat.modules.linear.__file__,}
```
I think it makes sense to trace through ```_prims``` and ```_refs```, but I didn't see the benefit of tracing ```ao``` builtin modules, can you elaborate more?

Reply:
As discussed offline, there doesn't seem to be a good way to work around the circular import issue. To unblock, we will land this change as is (maybe add some comments to explain). In the future, we should see if we can do something else other than expressing the file name through a string.